pkg/cmd/historical_logs.go

```diff
@@ -69,78 +86,166 @@ func NewCmdLogFilter(streams genericclioptions.IOStreams) *cobra.Command {
-	cmd.Flags().StringVar(&o.FinishTime, "finishtime", "", "Fetch Historical logs till a timestamp")
+	cmd.Flags().StringVar(&o.Tail, "tail", "", "Fetch Historical logs for the last N seconds, minutes, hours, or days")
 	cmd.Flags().StringVar(&o.Level, "level", "", "Fetch Historical logs from different logging level, Example: Info,debug,Error,Unknown, etc")
-	cmd.Flags().StringVar(&o.MaxLogs, "maxlogs", "", "Specify number of documents [logs] to be fetched. 1000 by default")
-	cmd.Flags().StringVar(&o.Index, "index", "", "Specify Index to filter by [infra-000001/app-000001/audit-000001")
-
+	cmd.Flags().StringVar(&o.Limit, "limit", "", "Specify number of documents [logs] to be fetched. 1000 by default")
+	cmd.Flags().StringVar(&o.Resources.DaemonSet, "daemonset", "", "Fetch logs from pods of a Daemon Set")
+	cmd.Flags().StringVar(&o.Resources.StatefulSet, "statefulset", "", "Fetch logs from pods of a Stateful Set")
+	cmd.Flags().StringVar(&o.Resources.Deployment, "deployment", "", "Fetch logs from pods of a Deployment")
+	cmd.Flags().BoolVar(&o.Prefix, "prefix", false, "Prefix each log with the log source (pod name and container name)")
 }
 
 func (o *LogParameters) Execute(streams genericclioptions.IOStreams) error {
 
-	kubeconfig := filepath.Join(
-		os.Getenv("HOME"), ".kube", "config",
-	)
+	kubernetesOptions, err := client.KubernetesClient()
+	if err != nil {
+		return err
+	}
+
+	err = o.ProcessLogParameters()
 
-	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
 	if err != nil {
-		return fmt.Errorf("Kubeconfig Error: %v", err)
+		return err
 	}
 
-	payload := new(bytes.Buffer)
-	err = json.NewEncoder(payload).Encode(o)
+	var podList []string
+
+	podList, err = k8sresources.GetResourcesPodList(kubernetesOptions, &o.Resources, o.Namespace)
 
 	if err != nil {
-		return fmt.Errorf("An Error Occurred while encoding JSON: %v", err)
+		return err
 	}
 
-	clusterUrl := config.Host
-	endIndex := strings.LastIndex(clusterUrl, ":")
-	startIndex := strings.Index(clusterUrl, ".") + 1
-	clusterName := clusterUrl[startIndex:endIndex]
+	endIndex := strings.LastIndex(kubernetesOptions.ClusterUrl, ":")
+	startIndex := strings.Index(kubernetesOptions.ClusterUrl, ".") + 1
+	clusterName := kubernetesOptions.ClusterUrl[startIndex:endIndex]
 
 	/*Example cluster URL : http://api.sangupta-tetrh.devcluster.openshift.com:6443. The first occurrence of '.' and last occurrence of ':'
 	act as start and end indices. Extract cluster name as substring using start and end Indices i.e, sangupta-tetrh.devcluster.openshift.com to build the log-exploration-api URL*/
 
-	ApiUrl := "http://log-exploration-api-route-openshift-logging.apps." + clusterName + "/logs/filter"
+	baseUrl := "http://log-exploration-api-route-openshift-logging.apps." + clusterName + "/logs/filter"
+
+	var logList []string
+
+	if len(podList) > 0 {
+		for _, pod := range podList {
+
+			o.Podname = pod
+
+			err := fetchLogs(&logList, baseUrl, o)
+			if err != nil {
+				return err
```

Comment:
nit: In case of failure we won't see logs from other pods. Maybe it makes sense to just print an error and continue to try to fetch logs from other pods.